util/queue.py

```diff
@@ -66,9 +71,20 @@ class Queue(multiprocessing.queues.Queue):
     """
 
     def __init__(self, *args, **kwargs):
+        if sys.version_info >= (3, 4) and 'ctx' not in kwargs:
+            kwargs['ctx'] = multiprocessing.get_context()
         super(Queue, self).__init__(*args, **kwargs)
         self._size = SharedCounter(0)
 
+    # __getstate__ and __setstate__ are needed for pickling, otherwise _size won't be copied.
```

Comment:
Let's drop this comment and instead add docstrings to the two methods, e.g. "Returns the contents to pickle for the instance" and "Sets the state of the instance upon unpickling".